File: MathsThing_Plugins/src/uk/ac/uea/mathsthing/plugins/functions/ASin.java
package uk.ac.uea.mathsthing.plugins.functions;

import java.math.BigDecimal;

import uk.ac.uea.mathsthing.IPlugin.IFunctionPlugin;
import uk.ac.uea.mathsthing.util.FormulaException;

public class ASin extends IFunctionPlugin {
	
	@Override
	public String getName()
	{
		return "asin";
	}
	
	@Override
	public BigDecimal function(BigDecimal input) throws FormulaException
	{
		return new BigDecimal(Math.asin(input.doubleValue()));
	}
}

File: MathsThing_Plugins/src/uk/ac/uea/mathsthing/plugins/functions/ASin.scala
package uk.ac.uea.mathsthing.plugins.functions

import uk.ac.uea.mathsthing.IPlugin.IFunctionPlugin
import java.math.BigDecimal

class ASin extends IFunctionPlugin {

  def getName : String = "asin"
    
  def function(input: BigDecimal) : BigDecimal = {
  	if (input.doubleValue == Double.NaN 
  	    || input.doubleValue <= Double.NegativeInfinity 
  	    || input.doubleValue >= Double.PositiveInfinity)
      null
    
    try {
      val result = new BigDecimal(Math.asin(input.doubleValue))
      
      if (result.doubleValue == Double.NaN 
  	      || result.doubleValue <= Double.NegativeInfinity 
  	      || result.doubleValue >= Double.PositiveInfinity 
  	      || result.doubleValue > 1000 || result.doubleValue < -1000)
        null;
  	  else
        result
    } catch {
      case ex:NumberFormatException => null
    }
  }
}